packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesModule.kt

```diff
@@ -255,36 +255,46 @@ class UpdatesModule(
   @ExpoMethod
   fun readLogEntriesAsync(maxAge: Long, promise: Promise) {
     AsyncTask.execute {
-      val reader = UpdatesLogReader()
+      val reader = UpdatesLogReader(context)
       val date = Date()
       val epoch = Date(date.time - maxAge)
-      val results = reader.getLogEntries(epoch).map {
-        val entry = UpdatesLogEntry.create(it)
-        Bundle().apply {
-          putLong("timestamp", entry.timestamp)
-          putString("message", entry.message)
-          putString("code", entry.code)
-          putString("level", entry.level)
-          if (entry.updateId != null) {
-            putString("updateId", entry.updateId)
-          }
-          if (entry.assetId != null) {
-            putString("assetId", entry.assetId)
-          }
-          if (entry.stacktrace != null) {
-            putStringArray("stacktrace", entry.stacktrace.toTypedArray())
+      val results = reader.getLogEntries(epoch)
+        .mapNotNull { UpdatesLogEntry.create(it) }
+        .mapNotNull { entry ->
+          Bundle().apply {
+            putLong("timestamp", entry.timestamp)
+            putString("message", entry.message)
+            putString("code", entry.code)
+            putString("level", entry.level)
+            if (entry.updateId != null) {
+              putString("updateId", entry.updateId)
+            }
+            if (entry.assetId != null) {
+              putString("assetId", entry.assetId)
+            }
+            if (entry.stacktrace != null) {
+              putStringArray("stacktrace", entry.stacktrace.toTypedArray())
+            }
           }
         }
-      }
       promise.resolve(results)
     }
   }
 
   @ExpoMethod
   fun clearLogEntriesAsync(promise: Promise) {
     AsyncTask.execute {
-      // This method is a no-op until persistent logs are implemented
-      promise.resolve(null)
+      val reader = UpdatesLogReader(context)
+      reader.purgeLogEntries(
+        olderThan = Date(),
+        {
+          if (it != null) {
+            promise.reject(it)
+          } else {
+            promise.resolve(null)
+          }
+        }
+      )
```

Comment:
(move lambda out of parentheses)

```suggestion
      reader.purgeLogEntries(
        olderThan = Date()
      ) { error ->
        if (error != null) {
          promise.reject(error)
        } else {
          promise.resolve(null)
        }
      }
```